src/common/copilot/user-feedback/CESSurvey.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import * as vscode from "vscode";
+import { npsAuthentication } from "../../../web/client/common/authenticationProvider";
+import { SurveyConstants } from "../../../web/client/common/constants";
+import fetch from "node-fetch";
+import { getNonce } from "../Utils";
+import { ITelemetry } from "../../../client/telemetry/ITelemetry";
+import { UserFeedbackFailureEvent, UserFeedbackSuccessEvent, sendTelemetryEvent } from "../telemetry/copilotTelemetry";
+
+
+export async function CESUserFeedback(context: vscode.ExtensionContext, sessionId: string, userID: string, thumbType: string, telemetry: ITelemetry) {
```

Comment:
can you please break this method into small ones. It is doing lot of functionalities.